src/opentype/tables/advanced/shared/subtables/lookuptypes/gsub.js

```diff
@@ -0,0 +1,29 @@
+import { ParsedData } from "../../../../../../parser.js";
+import { CoverageTable } from "../../coverage.js";
+
+export function undoCoverageOffsetParsing(instance) {
+  instance.parser.currentPosition -= 2;
+  delete instance.coverageOffset;
+  delete instance.getCoverageTable;
+}
+
+export class LookupType extends ParsedData {
+  constructor(p) {
+    super(p);
+    this.substFormat = p.uint16;
+    this.coverageOffset = p.Offset16;
+  }
+  getCoverageTable() {
+    let p = this.parser;
+    p.currentPosition = this.start + this.coverageOffset;
+    return new CoverageTable(p);
+  }
+}
+
+// used by types 5 and 6
+export class SubstLookupRecord {
+  constructor(p) {
+    this.glyphSequenceIndex = p.uint16; // Index into current glyph sequence — first glyph = 0.
+    this.lookupListIndex = p.uint16; // Lookup to apply to that position — zero-based index.
+  }
+}
```

Comment:
Like above, this would be:
Just for ease of reading, I prefer to use a single `export` statement at the end of the file rather than exporting each thing "with its declaration", so:
```
import { LookupType } from "./gsub.js";

function undoCoverageOffsetParsing ....

class LookupType extends ParsedData ...

class SubstLookupRecord  ...

export { undoCoverageOffsetParsing , LookupType, SubstLookupRecord }
```